src/audio/igo_nr/igo_nr.c

```diff
@@ -558,84 +484,52 @@ static int32_t igo_nr_set_chan(struct comp_dev *dev, struct sof_ipc_ctrl_data *c
 		}
 
 		if (val)
-			igo_nr_set_chan_process(dev, ch);
+			igo_nr_set_chan_process(cd, ch);
 		else
-			igo_nr_set_chan_passthrough(dev, ch);
+			igo_nr_set_chan_passthrough(cd, ch);
 	}
 
 	return 0;
 }
 
-static int32_t igo_nr_cmd_set_value(struct comp_dev *dev, struct sof_ipc_ctrl_data *cdata)
+static int igo_nr_set_config(struct processing_module *mod, uint32_t param_id,
+			     enum module_cfg_fragment_position pos,
+			     uint32_t data_offset_size, const uint8_t *fragment,
+			     size_t fragment_size, uint8_t *response,
+			     size_t response_size)
 {
-	int32_t ret;
+	struct comp_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
+	int ret;
 
-	switch (cdata->cmd) {
-	case SOF_CTRL_CMD_SWITCH:
-		comp_dbg(dev, "igo_nr_cmd_set_value(), SOF_CTRL_CMD_SWITCH, cdata->comp_id = %u",
-			 cdata->comp_id);
-		ret = igo_nr_set_chan(dev, cdata);
-		break;
+	comp_info(dev, "igo_nr_set_config()");
 
-	default:
-		comp_err(dev, "igo_nr_cmd_set_value() error: invalid cdata->cmd");
-		ret = -EINVAL;
-		break;
-	}
+	struct sof_ipc_ctrl_data *cdata = (struct sof_ipc_ctrl_data *)fragment;
 
-	return ret;
-}
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		comp_info(dev, "igo_nr_cmd_set_config(), SOF_CTRL_CMD_BINARY");
+		ret = comp_data_blob_set(cd->model_handler, pos, data_offset_size,
+					 fragment, fragment_size);
+		if (ret >= 0)
+			ret = igo_nr_check_config_validity(dev, cd);
 
-/* used to pass standard and bespoke commands (with data) to component */
-static int32_t igo_nr_cmd(struct comp_dev *dev,
-			  int32_t cmd,
-			  void *data,
-			  int32_t max_data_size)
-{
-	struct sof_ipc_ctrl_data *cdata = data;
+		return ret;
+	case SOF_CTRL_CMD_SWITCH:
+		comp_dbg(dev, "igo_nr_cmd_set_config(), SOF_CTRL_CMD_SWITCH, cdata->comp_id = %u",
+			 cdata->comp_id);
+		return igo_nr_set_chan(mod, cdata);
 
-	comp_info(dev, "igo_nr_cmd()");
-
-	switch (cmd) {
-	case COMP_CMD_SET_DATA:
-		return igo_nr_cmd_set_data(dev, cdata);
-	case COMP_CMD_GET_DATA:
-		return igo_nr_cmd_get_data(dev, cdata, max_data_size);
-	case COMP_CMD_SET_VALUE:
-		return igo_nr_cmd_set_value(dev, cdata);
-	case COMP_CMD_GET_VALUE:
-		return igo_nr_cmd_get_value(dev, cdata);
 	default:
-		comp_err(dev, "igo_nr_cmd() error: invalid command");
+		comp_err(dev, "igo_nr_cmd_set_config() error: invalid cdata->cmd");
 		return -EINVAL;
 	}
 }
 
-static void igo_nr_process(struct comp_dev *dev,
-			   struct comp_buffer *source,
-			   struct comp_buffer *sink,
-			   struct comp_copy_limits *cl,
-			   int32_t frames)
-
+static void igo_nr_print_config(struct processing_module *mod)
```

Comment:
this is an igo_nr specific function, possible to remove and add specific print to each place?

Reply:
Best to keep the classification in components as-is.  The owner of the component wanted to keep these traces as debug level.  We should fix our trace if debug traces can no more be activated from trace tool.